File: mindspore/lite/java/src/main/java/com/mindspore/Graph.java
/*
 * Copyright 2021-2022 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.mindspore;

import com.mindspore.config.MindsporeLite;

/**
 * Graph Class
 *
 * @since v1.0
 */
public class Graph {
    static {
        MindsporeLite.init();
    }

    private long graphPtr;

    /**
     * Construct function.
     */
    public Graph() {
        this.graphPtr = 0;
    }

    /**
     * Load file.
     *
     * @param file model file.
     * @return load status.
     */
    public boolean load(String file) {
        if (file == null) {
            return false;
        }
        this.graphPtr = loadModel(file);
        return this.graphPtr != 0L;
    }

    /**
     * Get graph pointer.
     *
     * @return graph pointer.
     */
    public long getGraphPtr() {
        return this.graphPtr;
    }

    /**
     * Fre graph pointer.
     */
    public void free() {
        this.free(graphPtr);
        graphPtr = 0;
    }

    private native long loadModel(String file);

    private native boolean free(long graphPtr);
}

File: mindspore/lite/src/extendrt/cxx_api/graph/graph.cc
/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#include "include/api/graph.h"
#include "cxx_api/graph/graph_data.h"
#include "utils/log_adapter.h"

namespace mindspore {
Graph::Graph() : graph_data_(nullptr) {}

Graph::Graph(const std::shared_ptr<GraphData> &graph_data) : graph_data_(graph_data) {}

Graph::Graph(std::shared_ptr<GraphData> &&graph_data) : graph_data_(graph_data) {}

Graph::~Graph() {}

Graph::Graph(std::nullptr_t) : graph_data_(nullptr) {}

bool Graph::operator==(std::nullptr_t) const { return graph_data_ == nullptr; }

bool Graph::operator!=(std::nullptr_t) const { return graph_data_ != nullptr; }

ModelType Graph::ModelType() const {
  MS_EXCEPTION_IF_NULL(graph_data_);
  return graph_data_->ModelType();
}
}  // namespace mindspore
